### PR TITLE
chore: update code review policy

### PR DESCRIPTION
**Pull request by igorlesnenko**
In this PR I propose an update to the code review policy in an attempt to reduce PR turnaround time.

- Added more focus on reviewing PR within 24 hours
- Added a suggestion to start working day from doing PR reviews
- Made "One review required" definition broader
- Added suggestion of requesting a review from a group of people or from everyone 
- Copied some info we have in notion to store everything in a single place

Please let me know what you think

### CODE REVIEW

File: docs/codeReview.md [hunk 1]
```diff
@@ -5,13 +5,27 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+- Consider requesting review from a group of reviewers at once
+- Post a link in a slack channel: [**#t_product_review_requests**](https://app.slack.com/client/T08FL6336/C043A70QXTL) to allow anyone interested to review this PR earlier than assigned Code Reviewer (unless you don't need one particular person)
+- If the PR has not been reviewed within 24 hours, send a reminder to Slack or request a review from another reviewer or group of reviewers, or consider adding an agenda item to the [Parabol – Product](https://www.notion.so/Parabol-Product-1065acffc95d4f64a71b7808bad98ff5) check-in meeting
+- If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. Use the label for PRs of any size and complexity If you are confident enough and it feels safe.
```
Comment (BartoszJarocki) [inline docs/codeReview.md:20]: -1 I'd add one exception here. We should be extremely careful with DB migrations as they might break the application in the way that is hard to restore.

File: docs/codeReview.md [hunk 2]
```diff
@@ -5,13 +5,27 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
-- It should be a reviewer’s goal to do the code review within 1 business day. If you’re unable to do it, leave a comment.
-- If you’re busy it’s ok to assign someone else as a reviewer.
 - When submitting a PR, provide context. What's the task? What's changed? Why? If you can, record a Loom to share why things are done that way. It'll be easier for a reviewer to understand what decision you've made and why.
+
+## Requesting Code Reviews
+
+- Self-assess which domains are involved in the changes made by your PR, find a reviewer from among the folks who fill the [Code Reviewer (L2–3)](https://www.notion.so/Code-Reviewer-L2-3-a47bb0759a0b41b5b0469ff14a8cdaae) role, paying specific attention to who GitHub suggests, who has indicated expertise within a domain (e.g. backend, front-end, etc.), and who has capacity to review within 24 hours.
+- Consider requesting review from a group of reviewers at once
+- Post a link in a slack channel: [**#t_product_review_requests**](https://app.slack.com/client/T08FL6336/C043A70QXTL) to allow anyone interested to review this PR earlier than assigned Code Reviewer (unless you don't need one particular person)
```
Comment (BartoszJarocki) [inline docs/codeReview.md:18]: +1 is this step necessary? Is there a way to post review requested events automatically to slack?

File: docs/codeReview.md [hunk 3]
```diff
@@ -5,13 +5,27 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+- If the PR has not been reviewed within 24 hours, send a reminder to Slack or request a review from another reviewer or group of reviewers, or consider adding an agenda item to the [Parabol – Product](https://www.notion.so/Parabol-Product-1065acffc95d4f64a71b7808bad98ff5) check-in meeting
+- If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. Use the label for PRs of any size and complexity If you are confident enough and it feels safe.
+
+## Doing Code Reviews
+
+- Doing code reviews and unblocking others is also important part of your job. It is totally fine that sometimes you may do more PR reviews than coding.
+- Prioritize unblocking others by starting working day by doing [code reviews](https://github.com/pulls/review-requested)
```
Comment (BartoszJarocki) [inline docs/codeReview.md:25]: +1 I'd change it to `by doing code reviews on daily basis`. Some folks may want to do it later in the day, not necessarily in the morning. let's keep that freedom

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]
A few notes, I'd say we should not allow usage of `one review required` with PRs containing DB migrations. Good stuff otherwise!

File: docs/codeReview.md (hunk 2 above)
Comment (nickoferrall) [reply docs/codeReview.md:18]: I agree, I think we can automate the Slack notification and remove this

**Review by nickoferrall — CHANGES_REQUESTED** [review]
This is really helpful!

File: docs/codeReview.md [hunk 4]
```diff
@@ -5,13 +5,27 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+- Try to complete PR reviews within 24 hours
```
Comment (Dschoordsch) [inline docs/codeReview.md:26]: +1 I would rephrase it to within 1 working day

File: docs/codeReview.md [hunk 5]
```diff
@@ -57,14 +71,4 @@ A PR advances from self-review to reviewer-review and finally maintainer review
...
-## One Review Required
-
-If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. If the Reviewer thinks it's not safe to have just one review, it's their responsiblity to add a comment saying that Maintainer review is required to merge the PR.
-
-Examples of PRs that could be safe to merge that way
-
-- Minor changes, e.g. [remove octokit](https://github.com/ParabolInc/parabol/pull/6479)
-- Small, non-architectural changes that happen behind a feature flag, e.g. [show GitLab issues in the Estimate phase](https://github.com/ParabolInc/parabol/pull/6355). It's the PR author's responsibility to keep track of the scenarios that need testing before removing the feature flag.
-- Work that is related to new features which are hidden after a feature flag and do not touch existing functionality (any work related to the architecture of the new feature is an exception and should go through the full review process) e.g [added end team prompt mutation](https://github.com/ParabolInc/parabol/pull/6250)
+- When you replied or resolved all comments, move the PR back to reviewer- or maintainer-review
```
Comment (Dschoordsch) [inline docs/codeReview.md:74]: @mattkrick are you still following the reviewer-review / maintainer-review columns or could we simplify this to just "review" and just follow `is:pr is:open user-review-requested:@me` search?

File: docs/codeReview.md (hunk 1 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: @BartoszJarocki added exception

File: docs/codeReview.md (hunk 2 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Removed

File: docs/codeReview.md (hunk 3 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Changed

File: docs/codeReview.md [hunk 6]
```diff
@@ -5,13 +5,27 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
 - Once you're assigned to a task, it's your responsibility to split it the way you can submit reasonably sized PRs. It’s ok to have multiple PRs for a single issue.
 - Use draft PRs to ask for early feedback.
-- It should be a reviewer’s goal to do the code review within 1 business day. If you’re unable to do it, leave a comment.
-- If you’re busy it’s ok to assign someone else as a reviewer.
 - When submitting a PR, provide context. What's the task? What's changed? Why? If you can, record a Loom to share why things are done that way. It'll be easier for a reviewer to understand what decision you've made and why.
+
+## Requesting Code Reviews
+
+- Self-assess which domains are involved in the changes made by your PR, find a reviewer from among the folks who fill the [Code Reviewer (L2–3)](https://www.notion.so/Code-Reviewer-L2-3-a47bb0759a0b41b5b0469ff14a8cdaae) role, paying specific attention to who GitHub suggests, who has indicated expertise within a domain (e.g. backend, front-end, etc.), and who has capacity to review within 24 hours.
+- Consider requesting review from a group of reviewers at once
```
[the remark this answers is not in the document]
Comment (igorlesnenko) [reply docs/codeReview.md]: Removed suggestion to request review from multiple persons

File: docs/codeReview.md (hunk 4 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Rephrased

File: docs/codeReview.md [hunk 7]
```diff
@@ -5,13 +5,25 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+- If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. Use the label for PRs of any size and complexity If you are confident enough, and it feels safe. One exception: we should be extremely careful with DB migrations as they might break the application in the way that is hard to restore.
+
+## Doing Code Reviews
+
+- Doing code reviews and unblocking others is also important part of your job. It is totally fine that sometimes you may do more PR reviews than coding.
+- Prioritize unblocking others by doing [code reviews](https://github.com/pulls/review-requested) on daily basis
+- Try to complete PR reviews within 1 working day
+- If not possible to complete PR review within 1 working day, follow the above “Requesting Code Reviews” guide to reassign the PR
+- If the review requested from a group of people, indicate that you are doing review by assigning a PR to yourself or in a comment.
```
Comment (nickoferrall) [inline docs/codeReview.md:26]: +1 I think this makes sense, but I favour requesting a review from one person instead of a group of people, so I'd prefer to remove this point as it suggests requesting a review from a group of people is an option

**Review by nickoferrall — APPROVED** [review]
LGTM!

**Review by BartoszJarocki — APPROVED** [review]
lgtm!

File: docs/codeReview.md (hunk 5 above)
Comment (mattkrick) [reply docs/codeReview.md:72]: that works for me! this trimester my exco responsibilities are preventing me from doing as much in the codebase as I'd like so I want to remove myself as a blocker as much as I can

File: docs/codeReview.md [hunk 8]
```diff
@@ -5,13 +5,25 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+
+## Requesting Code Reviews
+
+- Self-assess which domains are involved in the changes made by your PR, find a reviewer from among the folks who fill the [Code Reviewer (L2–3)](https://www.notion.so/Code-Reviewer-L2-3-a47bb0759a0b41b5b0469ff14a8cdaae) role, paying specific attention to who GitHub suggests, who has indicated expertise within a domain (e.g. backend, front-end, etc.), and who has capacity to review within 24 hours.
+- If the PR has not been reviewed within 24 hours, send a reminder to Slack or request a review from another reviewer or group of reviewers, or consider adding an agenda item to the [Parabol – Product](https://www.notion.so/Parabol-Product-1065acffc95d4f64a71b7808bad98ff5) check-in meeting
+- If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. Use the label for PRs of any size and complexity If you are confident enough, and it feels safe. One exception: we should be extremely careful with DB migrations as they might break the application in the way that is hard to restore.
+
+## Doing Code Reviews
```
Comment (mattkrick) [inline docs/codeReview.md:20]: +1 "Performing" over "Doing" for clarity (IMO "Do" is like "Get", it's one of those verbs that means a thousand things)

**Review by mattkrick — APPROVED** [review]

File: docs/codeReview.md [hunk 9]
```diff
@@ -5,13 +5,25 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
 - Use draft PRs to ask for early feedback.
-- It should be a reviewer’s goal to do the code review within 1 business day. If you’re unable to do it, leave a comment.
-- If you’re busy it’s ok to assign someone else as a reviewer.
 - When submitting a PR, provide context. What's the task? What's changed? Why? If you can, record a Loom to share why things are done that way. It'll be easier for a reviewer to understand what decision you've made and why.
+
+## Requesting Code Reviews
+
+- Self-assess which domains are involved in the changes made by your PR, find a reviewer from among the folks who fill the [Code Reviewer (L2–3)](https://www.notion.so/Code-Reviewer-L2-3-a47bb0759a0b41b5b0469ff14a8cdaae) role, paying specific attention to who GitHub suggests, who has indicated expertise within a domain (e.g. backend, front-end, etc.), and who has capacity to review within 24 hours.
+- If the PR has not been reviewed within 24 hours, send a reminder to Slack or request a review from another reviewer or group of reviewers, or consider adding an agenda item to the [Parabol – Product](https://www.notion.so/Parabol-Product-1065acffc95d4f64a71b7808bad98ff5) check-in meeting
```
Comment (jmtaber129) [inline docs/codeReview.md:17]: +1 
> or request a review from another reviewer or group of reviewers

I think I understand the intention behind this to mean "figure out if the reviewer is able to review your PR and then reassign review if they're not", but this reads a little like "if you don't feel like reminding them in Slack, just ping someone else". I also think adding an agenda item for a check-in that's more than a day away isn't effective.

Here's what I'd suggest here:
"If the PR has not been reviewed within 24 hours, ping the reviewer on Slack and ask when they'll be able to review. If they don't have capacity for review, reassign to a different reviewer. If you're still blocked at the next check-in meeting, add an agenda item."

**Review by jmtaber129 — APPROVED** [review]
LGTM, but one suggested change for steps a PR author should take if their review is taking a while.

**Review by JimmyLv — APPROVED** [review]
LGTM!

File: docs/codeReview.md (hunk 9 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Make sense! Replaced

File: docs/codeReview.md (hunk 8 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Make sense, replaced "Doing" everywhere

File: docs/codeReview.md (hunk 7 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Removed

File: docs/codeReview.md (hunk 5 above)
Comment (igorlesnenko) [reply docs/codeReview.md]: Changed to just "back to review"

File: docs/codeReview.md [hunk 10]
```diff
@@ -5,13 +5,25 @@ Review progress is tracked in the [Sprint Board](https://github.com/orgs/Parabol
...
+- If the developer creating the PR feels that maintainer review isn't necessary, they should add a label: “One Review Required”. Use the label for PRs of any size and complexity If you are confident enough, and it feels safe. One exception: we should be extremely careful with DB migrations as they might break the application in the way that is hard to restore.
```
[the remark this answers is not in the document]
Comment (igorlesnenko) [reply docs/codeReview.md]: Fixed
